### DISCUSSION TITLE
OGG / OGV

### DISCUSSION THREAD

**Discussion by dkloke** [aclap-dev/video-downloadhelper#672] [Help and Q&A]
### Your question

The OGG / OGV file format is not (well) supported. I'm not sure it's possible to get back the OGV format once the output has been streamed as H.264. The fact that the original file is stored as OGV on the remote resource doesn't mean that's the codec of the stream over the wire. Nonetheless, it would be nice to be able to use VH to capture the content, and then offer output as e.g., MP4 or FFMPEG- supported output.
 
I did download the content by previewing in FF, and context-menu Save Video As... which saved as .WEBM, sufficient for my needs. So this is not a critical issue.

I note that in the CoApp details it has "new_version":false, afaik I have installed the latest version from your site (2.0.19). Perhaps this flag is to prompt the user to update their coapp.
vdhcoapp-windows-x86_64-installer.exe
  CRC-32 : 933D4D7B
  SHA1 : F448B78B2C0B8D29A1E8B87DF0B09231193168D6
vdhcoapp.exe
  CRC-32 : 64661E33
  SHA1 : 614B2178AB67879ED66F646A770B52BCC4401E67.

I post this not-urgent-to-me bug report as suggested in the *gorgeous* and option-surfacing new v9 interface. Big thanks to Paul Rouget and the whole VideoHelper team! Even this little bug report was easy and smooth. This is how apps should be, thank you sincerely.

{
  "version": "9.0.1.13",
  "target": "mozilla",
  "channel": "stable",
  "page_url": "https://commons.wikimedia.org/wiki/File:Belt_Trick.ogv",
  "error": {
    "error": "coapp_failure",
    "details": "SideDownload error: : [in#0 @ 00000230b6377900] Error opening input: Invalid data found when processing input\r\nError opening input file https://commons.wikimedia.org/wiki/File:Belt_Trick.ogv.\r\nError opening input files: Invalid data found when processing input\r\n",
    "id": "coapp_failure_2519edb7-dac5-4e97-88a0-7e64948cadea",
    "downloadable_id": "downloadable_7440860766648382"
  }
}

### Addon & CoApp version and details

version: 9.0.1.13
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"ae1855faf70a4b4e"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "valid_license_message_has_been_displayed": true,
    "download_directory": "O:\\",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 2
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.1.22a1 beta, CoApp 2.0.19.

Interesting case.  When I just visited the Wikipedia page, I also got just an OGV in the VDH menu, & downloading it got the error you reported.

But then, I clicked the Play button on the video.  That opened the video in its own small window.  I played with the gear icon & selected the 1080p option.  After it completed playing & I closed the small window, back on the Wikipedia page I had this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/867c9427-0d74-4465-893b-d1a601b0ac61)

I downloaded the 10M WEBM.  This is what I got:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/7bdfb4d7-5caf-47f4-a50f-3bf5b1f5004d)

And here it is playing in VLC:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/b66e7552-9ca8-4c3e-b0bb-8557d0333805)

Then I thought to download that 19M OGV I'm showing on the VDH menu in the image above.  I got this:

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/866343e0-0109-48f5-83b0-ebc855201fd7)

Once I loaded the file into VLC, it properly detected the duration as 0:59.  I don't know what the deal is with the Windows properties saying it was over an hour.  Must be some AI bot doing a calculation using new math.  Of course, VLC played the OGV just fine.  You won't be surprised to learn that it was identical to the earlier WEBM.

So I'm not so sure there is really a bug here.  I suppose you could argue that if VDH puts something on its menu, it ought to be able to download it correctly.  But it's always been true that VDH sometimes needs you to launch playback on something before it will recognize it correctly.  That's just what happened here.  The object it recognized before launching playback is a bit of a mystery.  Maybe @paulrouget Paul has some blinding insights here.

**Reply by GrampaWildWilly** [reply]
Forgot to mention.  About this:

> CoApp\vdhcoapp.exe","version":"2.0.19","new_version":false}

I think it's just an internal indication of whether this is the first time you're executing a download that uses the CoApp.  There might be some one-time logic that gets executed on your first use.  But you know I'm guessing because, like I said, I don't work for the company, & I haven't read the code.  The code of the CoApp is publicly available, not that I flatter myself that I could make heads or tails of it.  The code of the VDH browser extension itself is not publicly available so that code isn't even available to me or any other mere mortal.  Once again, I doubt I could make heads or tails of it.

2.0.19 is the latest version & it's been around, near as I can tell, since 2024/2/19.  It's getting to be a real antique at a whole 3 months old.

In short, don't worry about this.
